src/Microsoft.AspNetCore.Razor.VSCode/src/CSharp/CSharpProjectedDocumentContentProvider.ts

```diff
@@ -3,28 +3,33 @@
  * Licensed under the MIT License. See License.txt in the project root for license information.
  * ------------------------------------------------------------------------------------------ */
 
-import * as vscode from 'vscode';
+import { IEventEmitterFactory } from '../IEventEmitterFactory';
 import { IRazorDocumentChangeEvent } from '../IRazorDocumentChangeEvent';
+import { IRazorDocumentManager } from '../IRazorDocumentManager';
 import { RazorDocumentChangeKind } from '../RazorDocumentChangeKind';
-import { RazorDocumentManager } from '../RazorDocumentManager';
 import { getUriPath } from '../UriPaths';
+import * as vscode from '../vscodeAdapter';
 
 export class CSharpProjectedDocumentContentProvider implements vscode.TextDocumentContentProvider {
     public static readonly scheme = 'virtualCSharp-razor';
 
-    private readonly onDidChangeEmitter = new vscode.EventEmitter<vscode.Uri>();
+    private readonly onDidChangeEmitter: vscode.EventEmitter<vscode.Uri>;
 
-    constructor(private readonly documentManager: RazorDocumentManager) {
+    constructor(
+        private readonly documentManager: IRazorDocumentManager,
+        eventEmitterFactory: IEventEmitterFactory) {
         documentManager.onChange((event) => this.documentChanged(event));
+        this.onDidChangeEmitter = eventEmitterFactory.create<vscode.Uri>();
     }
 
     public get onDidChange() { return this.onDidChangeEmitter.event; }
 
     public async provideTextDocumentContent(uri: vscode.Uri) {
         const razorDocument = this.findRazorDocument(uri);
         if (!razorDocument) {
-            vscode.window.showErrorMessage('For some reason the projected document isn\'t available.');
-            return;
+            // Document was removed from the document manager, meaning there's no more content for this
```

Comment:
Is there no other way to workaround this? Doesn't seem super right to not report an error when a document is missing.

Reply:
The only way would be to maintain a list of remove documents. Hmm, I'll add a trace level log here so if we encounter errors we can at least determine what happened.

Reply:
Sounds good 👍